flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/gsa/GatherSumApplyIteration.java

```diff
@@ -289,6 +300,11 @@ private GatherUdf(GatherFunction<VV, EV, M> gatherFunction, TypeInformation<Tupl
 
 		@Override
 		public void open(Configuration parameters) throws Exception {
+			try {
+				Collection<LongValue> numberOfVertices = getRuntimeContext().getBroadcastVariable("number of vertices");
+				this.gatherFunction.setNumberOfVertices(numberOfVertices.iterator().next().getValue());
+			} catch (Exception e) {
+			}
```

Comment:
Could also add a hasBroadcastVariable method to RuntimeContext. I had first expected that getBroadcastVariable would return an empty Collection.